x/gamm/pool-models/balancer/pool_suite_test.go

```diff
@@ -856,6 +856,79 @@ func (suite *KeeperTestSuite) TestCalcJoinPoolShares() {
 		})
 	}
 }
+func (suite *KeeperTestSuite) TestGetMaximalNoSwapLPAmount() {
+	type testcase struct {
+		name           string
+		shareOutAmount sdk.Int
+		swapFee        sdk.Dec
+		poolAssets     []balancer.PoolAsset
+		err            error
+	}
+
+	testcases := []testcase{
+		{
+			name:           "Share ratio is zero",
+			shareOutAmount: sdk.ZeroInt(),
+			swapFee:        sdk.ZeroDec(),
+			poolAssets: []balancer.PoolAsset{
+				{
+					Token:  sdk.NewInt64Coin("uosmo", 10),
+					Weight: sdk.NewInt(100),
+				},
+				{
+					Token:  sdk.NewInt64Coin("uatom", 10),
+					Weight: sdk.NewInt(100),
+				},
+			},
+			err: types.ErrInvalidMathApprox,
+		},
+		{
+			name:           "Share ratio is negative",
+			shareOutAmount: sdk.NewInt(-1),
+			swapFee:        sdk.ZeroDec(),
+			poolAssets: []balancer.PoolAsset{
+				{
+					Token:  sdk.NewInt64Coin("uosmo", 10),
+					Weight: sdk.NewInt(100),
+				},
+				{
+					Token:  sdk.NewInt64Coin("uatom", 10),
+					Weight: sdk.NewInt(100),
+				},
+			},
+			err: types.ErrInvalidMathApprox,
+		},
+		{
+			name:           "Pass",
+			shareOutAmount: sdk.NewInt(8_000_000_000_000_000_000),
+			swapFee:        sdk.ZeroDec(),
+			poolAssets: []balancer.PoolAsset{
+				{
+					Token:  sdk.NewInt64Coin("uosmo", 10),
+					Weight: sdk.NewInt(100),
+				},
+				{
+					Token:  sdk.NewInt64Coin("uatom", 10),
+					Weight: sdk.NewInt(100),
+				},
+			},
+		},
+	}
+	for _, tc := range testcases {
+		tc := tc
+
+		suite.T().Run(tc.name, func(t *testing.T) {
+			pool := createTestPool(t, tc.swapFee, sdk.ZeroDec(), tc.poolAssets...)
+			_, err := pool.GetMaximalNoSwapLPAmount(suite.Ctx, tc.shareOutAmount)
```

Comment:
let's also check returned values

Reply:
> 

It's based on this issue Mat, I don't know if this  issue still needs to solve

Reply:
Thanks your comment Mat, I updated check returned values 👍